From f7acf9fb53840d4ad00b3367b6c33977aa387d35 Mon Sep 17 00:00:00 2001
From: Michal Simek <michal.simek@xilinx.com>
Date: Thu, 28 Jan 2021 13:12:11 +0100
Subject: [PATCH 1757/1851] regulator: da9121: Init regmap to avoid compilation
 warning

commit 5a0168d04a7e7b4850666e35b6164208b4d92614 from
https://github.com/Xilinx/linux-xlnx.git

regmap is likely initialized all the time by below switch-case that's why
it is not causing incorrect code behavior.
That's why initialize regmap variable to suppress compilation warning.
Warning:
In file included from .../drivers/regulator/da9121-regulator.c:26:
.../drivers/regulator/da9121-regulator.c: In function 'da9121_i2c_probe':
../include/linux/regmap.h:844:27: warning: 'regmap' may be used
uninitialized in this function [-Wmaybe-uninitialized]
  844 |  __regmap_lockdep_wrapper(__devm_regmap_init_i2c, #config, \
      |                           ^~~~~~~~~~~~~~~~~~~~~~
/home/monstr/data/disk/linux/drivers/regulator/da9121-regulator.c:1249:24:
note: 'regmap' was declared here
 1249 |  struct regmap_config *regmap;
      |                        ^~~~~~

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/regulator/da9121-regulator.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/regulator/da9121-regulator.c b/drivers/regulator/da9121-regulator.c
index 5f56fa3c6a29..bc86e56a07ac 100644
--- a/drivers/regulator/da9121-regulator.c
+++ b/drivers/regulator/da9121-regulator.c
@@ -1246,7 +1246,7 @@ static int da9121_get_device_type(struct i2c_client *i2c, struct da9121 *chip)
 static int da9121_assign_chip_model(struct i2c_client *i2c,
 				    struct da9121 *chip)
 {
-	struct regmap_config *regmap;
+	struct regmap_config *regmap = NULL;
 	int ret = 0;
 
 	chip->dev = &i2c->dev;
-- 
2.31.1

